lib/news_scraper/trainer/uri_trainer.rb

```diff
@@ -2,12 +2,16 @@ module NewsScraper
   module Trainer
     class UriTrainer
       def initialize(uri)
-        @article_scrape_patterns = Transformers::Article.scrape_patterns
+        @article_scrape_patterns = YAML.load_file(NewsScraper::Constants::SCRAPE_PATTERN_CONFIG_FILE)
         @data_type = NewsScraper::Trainer::PresetsSelector.new(uri)
 
         @uri = uri
         @raw_html = Extractors::Article.new(uri: uri).extract
-        @transformer = Transformers::Article.new(uri: uri, payload: @raw_html)
+        @transformer = Transformers::Article.new(
```

Comment:
The issue I see with injecting scrape_pattern is that the client of this
interface knows way too much about the internal workings of
Transformers::Article.

Even injecting scrape_details is too much knowledge to know about for a
consumer.

Im thinking that we should create a separate transformer for the trainer
and abstract away the common logic in a mixin.

Ill poke at it when I get back.

On Aug 31, 2016 6:39 PM, "Julian Nadeau" <notifications@github.com> wrote:

> In lib/news_scraper/trainer/uri_trainer.rb
> <https://github.com/richardwu/news_scraper/pull/2#discussion_r77086560>:
>
> >          @data_type = NewsScraper::Trainer::PresetsSelector.new(uri)
> >
> >          @uri = uri
> >          @raw_html = Extractors::Article.new(uri: uri).extract
> > -        @transformer = Transformers::Article.new(uri: uri, payload: @raw_html)
> > +        @transformer = Transformers::Article.new(
>
> So, I didn't remove scrape patterns from the initializer yet, I think the
> transformers article will need a closer look at to remove it from there as
> scrape_details also relies on it.. its a bit of a smelly initializer tbh
> and the class deserves a bit of a refactor IMO
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/richardwu/news_scraper/pull/2/files/6b05d4a7834b6b4c73e91b60590cfa3d8ef8a9da#r77086560>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKEu8qkKaAQNMCDZwtcC0dH2uEwQUWVkks5qlgJjgaJpZM4JyDoO>
> .
>